src/App.tsx

```diff
@@ -2,14 +1,53 @@
+import React, { useMemo, useRef, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [showSuggestions, setShowSuggestions] = useState(false);
+  const [selectPerson, setSelectPerson] = useState<Person | null>(null);
+  const [highlightedIndex, setHighlightedIndex] = useState<number | null>(null);
+
+  const timer = useRef(0);
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(people =>
+      people.name.toLowerCase().includes(appliedQuery.toLowerCase()),
```

Comment:
Ensure that the filter function does not run if the input contains only spaces. You might want to trim the `appliedQuery` before checking its length.